### PR TITLE
Mesh element output

### PR DESCRIPTION
**Pull request by TomFischer**
Required for debugging of boundary conditions, especially for the element and node search routines.

### CODE REVIEW

File: MeshGeoToolsLib/convertMeshNodesToGeometry.cpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+#include "MeshGeoToolsLib/convertMeshNodesToGeometry.h"
+
+#include "MeshLib/Mesh.h"
+#include "MeshLib/Node.h"
+
+namespace MeshGeoToolsLib
+{
+void convertMeshNodesToGeometry(MeshLib::Mesh const& mesh,
```
Comment (rinkk) [inline MeshGeoToolsLib/convertMeshNodesToGeometry.cpp]: We already have MeshLib::convertMeshToGeo. It'd probably make sense to either move this functionality there or to move the other functionality here.

File: MeshLib/Utils/createMeshFromElements.h [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    {
+        new_elements.emplace_back(e->clone());
+    }
+
+   std::vector<MeshLib::Node*> new_nodes;
+    for (auto * e : new_elements)
+    {
+        for (std::size_t k(0); k < e->getNumberOfNodes(); ++k)
+        {
+            new_nodes.push_back(new MeshLib::Node(*(e->getNodes()[k])));
```
Comment (rinkk) [inline MeshLib/Utils/createMeshFromElements.h]: If you elements actually form a mesh you will get tons of duplicated nodes here. Not sure if this might be a problem.

File: MeshLib/Utils/createMeshFromElements.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+            e->setNode(k, new_nodes.back());
```
Comment (rinkk) [inline MeshLib/Utils/createMeshFromElements.h]: What happens to the old node? Isn't this a memory leak?

Comment (rinkk) [inline MeshLib/Utils/createMeshFromElements.h]: Sorry, it probably isn't a memory leak since the old element vector still exists...

**Comment by endJunction** [comment]
What is to be discussed/done here?

File: MeshLib/Utils/createMeshFromElements.h (hunk 2 above)
Comment (TomFischer) [inline MeshLib/Utils/createMeshFromElements.h]: This is true. The nodes will be cleaned up deploying `MeshLib::MeshRevision` object, see lines 46-48.

File: MeshGeoToolsLib/convertMeshNodesToGeometry.cpp (hunk 1 above)
Comment (TomFischer) [inline MeshGeoToolsLib/convertMeshNodesToGeometry.cpp]: Yes, you are right. I would move the existing `MeshLib::convertMeshToGeo` here `MeshGeoToolsLib` since `GeoLib` data structures as well as `MeshLib` data structures are needed. But the moving should be a separate PR.

**Comment by TomFischer** [comment]
Was thought as helper/debugger for balance process. Seems this is currently not to be needed anymore.
